frontend/src/main/java/com/google/tripmeout/frontend/storage/InMemoryPlaceVisitStorage.java

```diff
@@ -0,0 +1,107 @@
+package com.google.tripmeout.frontend.storage;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import com.google.common.collect.Tables;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.error.PlaceVisitAlreadyExistsException;
+import com.google.tripmeout.frontend.error.PlaceVisitNotFoundException;
+import com.google.tripmeout.frontend.error.TripNotFoundException;
+import com.google.tripmeout.frontend.storage.PlaceVisitStorage;
+import java.lang.RuntimeException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class InMemoryPlaceVisitStorage implements PlaceVisitStorage {
+  // <tripId, placeId, PlaceVisitModel>
+  Map<String, Map<String, PlaceVisitModel>> storage = new ConcurrentHashMap<>();
+
+  @Override
+  public void addPlaceVisit(PlaceVisitModel placeVisit)
+      throws PlaceVisitAlreadyExistsException, Throwable {
```

Comment:
don't throw `Throwable` especially in your interface, instead check if the cause is an instance of PlaceVisitAlreadyExistsException and cast it.

```java
  } catch (RuntimeException e) {
    if (e.getCause() instanceof PlaceVisitAlreadyExistsException) {
      throw (PlaceVisitAlreadyExistsException) e.getCause();
    }
    // unexpected cause, just re-throw the unchecked runtime exception.
    throw e;
  }
```